### PR TITLE
Add cli documentation

### PR DESCRIPTION
**Pull request by pdelvo**
This was requested in #806.

### CODE REVIEW

File: documentation/DotNetCli.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      "version": "1.0.0",
+      "type": "build"
+    }
+```
+
+If the project is restored and built right now this will already run the analyzers. A few extra steps are needed if they should be configured.
+
+## Rulesets and stylecop.json
+
+To supply a ruleset file and a stylecop.json configuration file to the compiler they have to be manually added as arguments to the compiler. For this add the follwing under the `compilationOptions` node in the project.json file:
```
Comment (vweijsters) [inline documentation/DotNetCli.md]: Small typo: `follwing` should be `following`

**Comment by vweijsters** [comment]
:+1: looks good to me besides the small typo

File: documentation/DotNetCli.md (hunk 1 above)
Comment (pdelvo) [inline documentation/DotNetCli.md]: Fixed

File: documentation/DotNetCli.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+If the project is restored and built right now this will already run the analyzers. A few extra steps are needed if they should be configured.
+
+## Rulesets and stylecop.json
+
+To supply a ruleset file and a stylecop.json configuration file to the compiler they have to be manually added as arguments to the compiler. For this add the following under the `compilationOptions` node in the project.json file:
+```json
+    "additionalArguments": [ "/ruleset:path/to/ruleset.ruleset", "/additionalfile:path/to/stylecop.json" ]
+```
+
+## Enabling xml documentation processing
```
Comment (sharwell) [inline documentation/DotNetCli.md]: :bulb: Consider merging this section with the documentation for [SA0001](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA0001.md).

**Comment by igatanasov** [comment]
compilationOptions is obsolete, should we use buildOptions and which argument?

**Comment by sharwell** [comment]
@igatanasov I would recommend creating a new issues for that concern. :+1:

**Comment by igatanasov** [comment]
Thanks @sharwell ,

I have just replied to this #806

**Pull request merged**